src/mame/sega/sega_beena.cpp

```diff
@@ -593,7 +607,7 @@ void sega_9h0_0008_state::video_reg_w(offs_t offset, uint32_t data, uint32_t mem
 
 uint32_t sega_9h0_0008_state::midi_reg_r(offs_t offset)
 {
-	if (offset == 0 && m_midi_busy_count > 0) {
+	if (!machine().side_effects_disabled() && offset == 0 && m_midi_busy_count > 0) {
 		m_midi_busy_count--;
 		return 0x2;
 	}
```

Comment:
It might be better to change this to:
```c++
	if (offset == 0 && m_midi_busy_count > 0) {
		if (!machine().side_effects_disabled())
			m_midi_busy_count--;
		return 0x2;
	}
```
Then it will correctly show the next value that will be read.